### PR TITLE
Move projects out of config.xml

### PR DESCRIPTION
**Pull request by brubo1**
Added Project table and project front-end.

### CODE REVIEW

File: modules/configuration/php/NDB_Form_configuration.class.inc [hunk 1]
```diff
@@ -106,6 +106,42 @@ class NDB_Form_Configuration extends NDB_Form
         $this->tpl_data['sandbox']         = $config->getSetting("sandbox");
 
     }
+
+    function project() {
+        $factory = NDB_Factory::singleton();
+        $config = $factory->config();
+
+        $projectList = Utility::getprojectList();
```
Comment (christinerogers) [inline modules/configuration/php/NDB_Form_configuration.class.inc]: This code should check the config setting useProject before calling Utility:getprojectList().  If useProject is false, getprojectList() will return an error (not just a null array)

**Comment by taracampbell** [comment]
useProject setting is stored in the database config tables. You can alter it from the front end in the config module.

File: modules/configuration/php/NDB_Form_configuration.class.inc [hunk 2]
```diff
@@ -106,6 +106,47 @@ class NDB_Form_Configuration extends NDB_Form
         $this->tpl_data['sandbox']         = $config->getSetting("sandbox");
 
     }
+
+    function project() {
+        $factory = NDB_Factory::singleton();
+        $config = $factory->config();
+	
+	/*if($config->getSetting(useProject)){*/
+	    $projectList = Utility::getprojectList();
```
Comment (christinerogers) [inline modules/configuration/php/NDB_Form_configuration.class.inc]: should be getProjectList(), no? (capital P)

Comment (brubo1) [inline modules/configuration/php/NDB_Form_configuration.class.inc]: About the useProject. Is this function supposed to give all the projects of every studies or give all the projects of that study?

**Comment by driusan** [comment]
Can you add a change to this to trigger Travis? It seems to be working on new changes again.

**Comment by brubo1** [comment]
Right away

**Comment by brubo1** [comment]
Now the errors are warnings. Both warnings are lines which have above 85 characters.

**Comment by taracampbell** [comment]
Hi Bruno, you need to run phpcs on the files. And fix any issues that come up. If the lines are too long, you can split them onto multiple lines.

**Comment by brubo1** [comment]
Integration tests are not running again.

**Comment by driusan** [comment]
Tests aren't running because it's has conflicts and isn't mergeable. Can you rebase it? (Both to fix that, and because the subproject frontend is merged and shouldn't show up in this diff..)

**Comment by brubo1** [comment]
I get this error at line 2061.
FAILURES!
Tests: 3, Assertions: 4, Errors: 1.
The command "vendor/bin/phpunit modules/new_profile/test" exited with 2.
This error keeps repeating itself and I don't know what to do with it.

**Comment by taracampbell** [comment]
Hey Bruno, another thing you might consider doing is a rebase to squash these commits. I can show you how to do this next week if you are unsure of how to do this.

**Comment by brubo1** [comment]
Hey Tara,

That would be nice. Thank you

Bruno

**Comment by brubo1** [comment]
Hello, this pull request can now be closed. I put the changes in another pull request. The name of the pull request is the same as this one.

**Pull request closed without merging**